fw/pool.c

```diff
@@ -190,37 +190,52 @@ tfw_pool_free(TfwPool *p, void *ptr, size_t n)
 	}
 }
 
+/**
+ * Delete single chunk that holds @ptr.
+ */
+void
+tfw_pool_clean_single(TfwPool *pool, void *ptr)
+{
+	TfwPoolChunk *c, *next, *prev = pool->curr;
+
+	BUG_ON(!pool);
+	BUG_ON(!ptr);
+
+	for (c = pool->curr->next; c; c = next) {
+		if (!(next = c->next))
```

Comment:
Why we can't delete the last chunk if ptr belong to it?

Reply:
No, we must not delete current and must not delete head of the pool, that located in the last chunk. I have a lot of comments about it in tests.